src/cca/scripts/cordova_core_connector.ts

```diff
@@ -29,15 +29,22 @@ export var uProxyAppChannel = freedom(
   return uProxyModuleFactory();
 });
 
-export function MakeCoreConnector() : CoreConnector {
-  let browserConnector = new CordovaCoreConnector({name: 'uproxy-ui-to-core-connector'});
-  return new CoreConnector(browserConnector);
+export function MakeCoreConnector() : Promise<CoreConnector> {
+  return uProxyAppChannelPromise.then((channel) => {
+    let browserConnector = new CordovaCoreConnector(
```

Comment:
I think this would be a little better:

```ts
const browserConnector = new CordovaCoreConnector(channel, {
  name: 'uproxy-ui-to-core-connector'
});
```